cocotb/types/logic_array.py

```diff
@@ -115,12 +120,38 @@ class LogicArray(Array[Logic]):
 
     __slots__ = ()
 
+    @typing.overload
     def __init__(
         self,
         value: typing.Union[int, typing.Iterable[LogicConstructibleT], BinaryValue],
+        range: typing.Optional[Range],
```

Comment:
nit for a future cleanup: It's rather common to import the used types from typing to use e.g. "Optional" instead of "typing.Optional". The type annotations are already long enough, prefixing them with "typing." isn't helping.